core/vm/contracts.go

```diff
@@ -410,6 +410,10 @@ func (c *bigModExp) Run(input []byte) ([]byte, error) {
 		expLen  = new(big.Int).SetBytes(getData(input, 32, 32)).Uint64()
 		modLen  = new(big.Int).SetBytes(getData(input, 64, 32)).Uint64()
 	)
+	// Check that all inputs are `u256` (32 - bytes), revert otherwise
+	if baseLen != 32 || expLen != 32 || modLen != 32 {
+		return nil, ErrExecutionReverted
```

Comment:
Should we return `ErrExecutionReverted` or some specific error as in https://github.com/scroll-tech/go-ethereum/pull/337

Reply:
As far as I know, the difference is that `ErrExecutionReverted` uses up all the gas provided to the call context, while other errors do not.

https://github.com/scroll-tech/go-ethereum/blob/8057f78edf121bba83b9f503bb6d1073d12f1d41/core/vm/evm.go#L244-L246

I think a modexp-specific error might be better here. What do you think? @zimpha

Reply:
>  `ErrExecutionReverted` uses up all the gas provided to the call context, while other errors do not.
Yes, that is why I am asking

Reply:
Since we don't provide a reason for the revert, it's better to return a specific error.